example/lib/main.dart

```diff
@@ -79,8 +83,18 @@ class _MyAppState extends State<MyApp> with SingleTickerProviderStateMixin {
     Airship.onShowInbox
         .listen((event) => debugPrint('Show inbox'));
 
-    Airship.onShowInboxMessage
-        .listen((messageId) => debugPrint('Show inbox message $messageId'));
+    Airship.onShowInboxMessage.listen((messageId){
+      Airship.inboxMessages.then((List<InboxMessage> messages) {
+        InboxMessage toShow = messages.firstWhere((thisMessage) =>
+        messageId == thisMessage.messageId,
+            orElse: () => null);
+
+        key.currentState.push(MaterialPageRoute<Null>(builder: (BuildContext context) {
+          return MessageView(message: toShow,);
```

Comment:
what happens when toShow is null?

Reply:
I don't think there's a very strong likelihood that the callback will be called with a message id that isn't in the list, but it would result in big red screen/NoSuchMethodError if it happened. Wrapped it in a check.